Logger/BaseLoggerMixins.cs

```diff
@@ -2,6 +2,25 @@
 {
     public static class BaseLoggerMixins
     {
-        
+        public static void Error(this BaseLogger bLogger, string message,params string[] args)
+        {
+            if (bLogger == null) { throw new System.ArgumentNullException("The BaseLogger Parameter is null"); }
+            bLogger.Log(LogLevel.Error, message);
+        }
+        public static void Debug(this BaseLogger bLogger, string message, params string[] args)
```

Comment:
From the coding guidelines

> DO NOT use abbreviations or contractions within identifier names.

```suggestion
        public static void Debug(this BaseLogger baseLogger, string message, params string[] args)
```